### PR TITLE
Remove task max retry count config

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
- Remove task max retry count config and use a fixed value of 30.

**Why?**
- The config is only used for determine when special task processing error log and task attempt metric should be emitted, not actually the maximum number of attempts. So removing this config and a fixed value should be good enough for this use case.

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: service/history/queues/executable.go [hunk 1]
```diff
@@ -90,6 +89,9 @@ const (
 	// resourceExhaustedResubmitMaxAttempts is the same as resubmitMaxAttempts but only applies to resource
 	// exhausted error
 	resourceExhaustedResubmitMaxAttempts = 1
+	// taskCriticalAttempts, if exceeded, task attempts metrics and critical processing error log will be emitted
+	// while task is retrying
+	taskCriticalAttempts = 30
```
Comment (alexshtin) [inline service/history/queues/executable.go:94]: I would call it more explicitly: `skipLogFirstTaskFailureAttempts` or something like this.

File: service/history/queues/executable.go [hunk 2]
```diff
@@ -209,7 +208,7 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 			defer e.Unlock()
 
 			e.attempt++
-			if e.attempt > e.criticalRetryAttempt() {
+			if e.attempt > taskCriticalAttempts {
```
Comment (alexshtin) [inline service/history/queues/executable.go:211]: Log message is bad. "Critical" is already in the tag and number of attempts is missing (but might be useful):
```
e.logger.Error("Error processing task after number of attempts.", tag.Error(err), tag.Attempt(e.attempt), tag.OperationCritical)
```

File: common/dynamicconfig/constants.go [hunk 3]
```diff
@@ -432,8 +430,6 @@ const (
 	TransferProcessorMaxPollRPS = "history.transferProcessorMaxPollRPS"
 	// TransferProcessorMaxPollHostRPS is max poll rate per second for all transferQueueProcessor on a host
 	TransferProcessorMaxPollHostRPS = "history.transferProcessorMaxPollHostRPS"
-	// TransferTaskMaxRetryCount is max times of retry for transferQueueProcessor
-	TransferTaskMaxRetryCount = "history.transferTaskMaxRetryCount"
```
Comment (alexshtin) [inline common/dynamicconfig/constants.go:436]: This was very confusing name!

**Review by alexshtin — APPROVED** [review]

File: service/history/queues/executable.go (hunk 2 above)
Comment (yycptt) [reply service/history/queues/executable.go:211]: I am not sure how useful # of attempts is in the log given that we have the attempt count metric. But I can add it.
 
Re. the message itself, I kinda agree. My main concern is oncall needs to memorize a new way for finding this important log (this message is used/searched a lot). So if it's not something you have strong opinion on, I'd like to leave it as is.

**Pull request merged**